daliuge-engine/dlg/manager/session.py

```diff
@@ -97,6 +98,30 @@ def __init__(self, sessionId, nm=None):
         self._nm = nm
         self._dropsubs = {}
 
+        class SessionFilter(logging.Filter):
+            def __init__(self, sessionId):
+                self.sessionId = sessionId
+
+            def filter(self, record):
+                if hasattr(record, 'session_id'):
+                    return record.session_id == self.sessionId
+                return False
```

Comment:
Maybe simpler to use `return getattr(record, 'session_id', None) == self.sessionId`

Reply:
Done